packages/sitecore-jss/src/layout-service.test.ts

```diff
@@ -200,6 +200,58 @@ describe('LayoutService', () => {
         });
     });
 
+    it('should catch 404 when request layout data', () => {
+      mock.onGet().reply(() => {
+        return [
+          404,
+          {
+            sitecore: {
+              context: {
+                pageEditing: false,
+                language: 'en',
+              },
+              route: null,
+            },
+          },
+        ];
+      });
+
+      const service = new RestLayoutService({
+        apiHost: 'http://sctest',
+        apiKey: '0FBFF61E-267A-43E3-9252-B77E71CEE4BA',
+        siteName: 'supersite',
+      });
+
+      return service.fetchLayoutData('/styleguide', 'en').then((layoutServiceData: any) => {
+        expect(layoutServiceData).to.deep.equal({
+          sitecore: {
+            context: {
+              pageEditing: false,
+              language: 'en',
+            },
+            route: null,
+          },
+        });
+      });
+    });
+
+    it('should throw error when request layout data', () => {
```

Comment:
This test name is a little unclear. Maybe something like this?
```suggestion
    it('should allow non 404 errors through', () => {
```

Reply:
Looks good. Done